### PR TITLE
filter forbidden

### PR DESCRIPTION
**Pull request by TA2k**
https://github.com/iobroker-community-adapters/ioBroker.openknx/issues/40

### CODE REVIEW

File: lib/tools.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable quotes */
 const axios = require("axios").default;
 const DPTLib = require(__dirname + '/knx/src/dptlib'); //todo for the moment
-
+const FORBIDDEN_CHARS = /[^._\-/ :!#$%&()+=@^{}|~\p{Ll}\p{Lu}\p{Nd}]+/gu;
```
Comment (boellner) [inline lib/tools.js:4]: question: is this documented in iob?

File: lib/projectImport.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ module.exports = {
 
             //format and put each found group address into the object tree
             _.each(groupAddresses, function (groupAddress) {
-                const gaName = groupAddress.getAttribute("Name");
+                const gaName = groupAddress.getAttribute("Name").replace(tools.FORBIDDEN_CHARS, '_');
                 let autoread = true;
                 let dptObj = {};
                 let fullPath = tools.formatGa(gaName);
```
Comment (boellner) [inline lib/projectImport.js:47]: assuming mittelgruppe, hauptgruppe .. can also have this problem, not only ga names

it needs a unified rule for gaName and tools.formatGa

**Review by boellner — CHANGES_REQUESTED** [review]
see code

File: lib/tools.js (hunk 1 above)
Comment (TA2k) [reply lib/tools.js:4]: Not directly
https://github.com/ioBroker/ioBroker.js-controller/blob/2bedca9238546950bae7682062419cd3ec7954c6/packages/common/lib/common/tools.js#L43

Comment (boellner) [reply lib/tools.js:4]: ok

File: lib/projectImport.js (hunk 2 above)
Comment (TA2k) [reply lib/projectImport.js]: moved filtering into formatting

File: lib/tools.js [hunk 3]
```diff
@@ -131,6 +131,9 @@ function isEmptyObject(obj) {
  * iobroker does not support all characters, substitte by _
  */
 function formatGa(gaName) {
+    //filter forbidden characters
+    gaName = gaName.replace(FORBIDDEN_CHARS, '_');
+    //format gaName
     return gaName.replace(/[\.\s\/]/g, '_');
```
Comment (boellner) [inline lib/tools.js:137]: can then be removed

**Review by boellner — CHANGES_REQUESTED** [review]

File: lib/tools.js (hunk 3 above)
Comment (TA2k) [reply lib/tools.js:137]: What exactly?
formatting? you replacing space with _ but space is not a forbidden char

Comment (boellner) [reply lib/tools.js:137]: ok

**Review by boellner — APPROVED** [review]

**Pull request merged**